registrations/permissions.py

```diff
@@ -57,21 +57,28 @@ def has_permission(self, request: Request, view: APIView) -> bool:
 
     @staticmethod
     def _has_object_update_permission(request: Request, obj: SignUp) -> bool:
-        if request.user.is_superuser or request.user.is_registration_admin_of(
-            obj.publisher
+        if (
+            request.user.is_superuser
+            or request.user.is_registration_admin_of(obj.publisher)
+            # Registration user who created the signup can update all the fields
+            or (
+                request.user.is_registration_user_access_user_of(
+                    obj.registration.registration_user_accesses
+                )
+                and obj.created_by_id == request.user.id
+            )
         ):
             return True
 
-        if obj.registration.registration_user_accesses.filter(
-            email=request.user.email
-        ).exists():
-            # Registration user can only change presence_status and nothing else,
-            # and they need to be strongly identified.
+        if request.user.is_registration_user_access_user_of(
+            obj.registration.registration_user_accesses
+        ):
+            # Strongly identified registration user can only change presence_status
+            # and nothing else.
             data_keys = set(request.data.keys())
-            return request.user.is_strongly_identified and (
-                data_keys == {"id", "presence_status"}
-                or data_keys == {"presence_status"}
-            )
+            return data_keys == {"id", "presence_status"} or data_keys == {
```

Comment:
Now the `is_registraton_user_access_user_of` check will be done twice if `request.user` is a registration user, but not the `created_by` user. To prevent this, I would maybe check `obj.created_by_id == request.user.id` here:
```
return obj.created_by_id == request.user.id or (data_keys == <x> or data_keys == <y>)
```